alpaca_trade_api/rest.py

```diff
@@ -220,16 +243,41 @@ def list_orders(self, status=None, limit=None, after=None, until=None,
         resp = self.get(url, params)
         return [Order(o) for o in resp]
 
-    def submit_order(self, symbol, qty, side, type, time_in_force,
-                     limit_price=None, stop_price=None, client_order_id=None,
-                     extended_hours=None, order_class=None,
-                     take_profit=None, stop_loss=None):
-        '''Request a new order'''
+    def submit_order(self,
+                     symbol: str,
+                     qty: int,
+                     side: str,
+                     type: str,
+                     time_in_force: str,
+                     limit_price: str = None,
+                     stop_price: str = None,
+                     client_order_id: str = None,
+                     extended_hours: bool = None,
+                     order_class: str = None,
+                     take_profit: dict = None,
+                     stop_loss: dict = None):
+        """
+        :param symbol: symbol or asset ID
+        :param qty: int
+        :param side: buy or sell
+        :param type: market, limit, stop, or stop_limit
+        :param time_in_force: day, gtc, opg, cls, ioc, fok
+        :param limit_price: str of float
```

Comment:
I believe lots of users put some number type here. Does it break such code or not?

Reply:
In this cas, str is preferred but nothing prevents int/float. What would be the right hinting?

Reply:
general guidelines should always be to give one way of operation. all users call the apis in the same way. it prevents a lot of headache in example code, documentation, errors and different behavior in different use cases(in my opinion) 
so the hints should be as such.
having said that - typings are not enforced, a user could decide for instance to pass a float here. the interpreter will allow it. 
the typings are used to help all users write the code in the same way (a guideline)